contributors/adding-dependencies.md

```diff
@@ -0,0 +1,129 @@
+3. Select the *additional (Phovea) modules* that should be included
+4. Confirm your selection with Enter and move the next step
+5. Select *additional libraries* (e.g., jQuery, D3) that are known by the Phovea framwork
+6. Confirm your selection with Enter
+7. Override files if you will be asked to
+8. (Optional, but recommended: Commit the changes to your application / plugin repository)
+9. [Update your setup](#update-setup)
+
+<a id="add-external-libs"></a>
+
+### Add other external libraries
+
+Follow theses steps only, if you cannot find the library in the known libraries by the Phovea framework (using the Phovea generator in the step above).
+
+1. Research if the library and the *@types* (TypeScript typings) are available on [npmjs](https://www.npmjs.com/)
+2. If yes, continue. If not, ask the library author to publish the library on npmjs.
+3. Open the *package.json* of the application/plugin repository
+4. Add the library and typings in the corresponding versions to the [dependencies section](https://docs.npmjs.com/files/package.json#dependencies). For example:
+
+  ```json
+  "dependencies": {
+    "@types/d3": "3.5.36",
+    "d3": "3.5.17"
+  },
+  ```
+   
+5. [Update your setup](#update-setup)
+
+
+### Version conflicts with D3 (v3 vs. v4)
+
+Most of the Phovea repositories (e.g., [phovea_vis](https://github.com/phovea/phovea_vis/) and [phovea_clue](https://github.com/phovea/phovea_clue)) use D3 v3. If you want to use code that uses D3 v4 add the specific D3 module (e.g., [d3-selection](https://github.com/d3/d3-selection)) and the @type dependency (as [described above](#add-external-libs)) to the *package.json*.
+
+```json
+"dependencies": {
+   "@types/d3-scale": "^1.0.4",
+   "d3-scale": "^1.0.3"
+},
+```
+
+Note, that you must import the specific D3 v4 modules to your TypeScript file as follows to avoid conflicts with the D3 v3 library:
+
+```js
+import {scaleLinear} from 'd3-scale';
+```
+
+-----
+
+<a id="server-dependencies"></a>
+
+## Adding server-side dependencies
+
+A server-side dependency can be another Phovea module or pip package available on [PyPI Python Package Index](https://pypi.python.org/pypi).
+
+### Phovea server plugins
+
+1. Open the command line and navigate to the directory of the server library 
+2. Run `yo phovea:add-dependency`
+3. Select the *additional (Phovea) modules* that should be included
+4. Confirm your selection with Enter and move the next step
+5. Select *additional libraries* (e.g., MongoDB, Redis) that are known by the Phovea framwork
+6. Confirm your selection with Enter
+7. Override files if you will be asked to
+8. (Optional, but recommended: Commit the changes to your application / plugin repository)
+9. [Update your setup](#update-setup)
+
+
+### pip dependencies
+
+Follow theses steps only, if you cannot find a suitable Phovea plugin (using the Phovea generator in the step above).
+
+1. Navigate to the server library directory
+2. Open the *.yo-rc.json*
```

Comment:
Is the alternative then to add the dependency directly to the *requirements.txt*?